pyrometheus/chem_expr.py

```diff
@@ -164,8 +167,63 @@ def _zeros_like(argument):
 
 # }}}
 
-# {{{ Equilibrium constants
 
+# {{{ Transport polynomials & mixture rules
+
+def transport_polynomial_expr(c, n, t):
+    """Generate code for transport polynomials
+
+    :returns: Transport polynomial expression with coefficients c in terms of
+    the temperature t as a :class:`pymbolic.primitives.Expression`. For
+    `thermal_conductivity`, `n = 1`, while for `viscosity` `n = 2`
+    """
+    assert len(c) == 5
+    return (
+        (
+            c[0]
+            + c[1] * p.Variable("log")(t)
+            + c[2] * p.Variable("log")(t) ** 2
+            + c[3] * p.Variable("log")(t) ** 3
+            + c[4] * p.Variable("log")(t) ** 4
+        )**n
+    )
+
+
+def viscosity_mixture_rule_wilke_expr(sol: ct.Solution, sp, x, mu):
+    """Generate code for species mixture rule.
+       See Robert J. Kee, Michael E. Coltrin and Peter Glarborg
+       "Chemically Reacting Flow" book, chapter 12.
+
+    :returns: Expression for the Wilke viscosity mixture rule
+        for species *sp* in terms of species mole fractions *w*
+        and viscosities *mu* as a :class:`pymbolic.primitives.Expression`
+    """
+    w = sol.molecular_weights
+    sqrt = p.Variable("sqrt")
+    return sum([x[j]*(
+        1 + sqrt((mu[sp]/mu[j])*sqrt(w[j]/w[sp]))
+    )**2 / sqrt(
+        8*(1 + (w[sp]/w[j]))
+    ) for j in range(sol.n_species)])
+
+
+def species_mass_diff_mixture_rule_expr(sol: ct.Solution, sp, mmw, x, diff_ij):
+    """Generate code for species mixture rule.
+       See Robert J. Kee, Michael E. Coltrin and Peter Glarborg
+       "Chemically Reacting Flow" book, chapter 12.
```

Comment:
Fix indentation. (Should line up with rest of code.)

Reply:
Removed these two lines and added the proper citation.